examples/tests/microbit_v2/main.c

```diff
@@ -0,0 +1,163 @@
+    printf("Function: %s returned error: %s\n", func_name, tock_strrcode(retval));
+    tock_exit(1);
+  }
+  return retval;
+}
+
+
+static bool tone_complete = true;
+static void tone_callback(void) {
+  // allow additional notes to be played
+  tone_complete = true;
+}
+
+static void light_leds(int leds[], int len) {
+  for (int i = 0; i < len; i++) {
+    check_err(led_on(leds[i]), "led_on");
+  }
+}
+
+static void button_callback(int                            btn_num,
+                            int                            button_pressed,
+                            __attribute__ ((unused)) int   arg2,
+                            __attribute__ ((unused)) void *ud) {
+
+  // Do not run if this was a button release or if tone is not finished
+  if (!button_pressed || !tone_complete) {
+    return;
+  }
+
+  // Play a note and make an LED pattern appear
+  tone_complete = false;
+  if (btn_num == 0) {
+    check_err(tone(NOTE_C4 * 3, 500, tone_callback), "tone");
+    int leds[10] = {2, 6, 8, 11, 12, 13, 16, 18, 21, 23};
+    light_leds(leds, 10);
+  } else if (btn_num == 1) {
+    check_err(tone(NOTE_E4 * 3, 500, tone_callback), "tone");
+    int leds[11] = {1, 2, 6, 8, 11, 12, 13, 16, 18, 21, 22};
+    light_leds(leds, 11);
+  } else if (btn_num == 2) {
+    check_err(tone(NOTE_G4 * 3, 500, tone_callback), "tone");
+    int leds[9] = {0, 1, 2, 3, 4, 7, 12, 17, 22};
+    light_leds(leds, 9);
+  }
+}
+
+static void sample_sensors(void) {
+  check_err(led_toggle(0), "led_toggle");  // Blink top left LED
+  for (int i = 1; i < 25; i++) {
+    check_err(led_off(i), "led_off");
+  }
+
+  // Sensors: temperature, acceleration, sound pressure
+  int temp = 0;
+  check_err(temperature_read_sync(&temp), "temperature_read_sync");
+  uint32_t accel_mag = check_err(ninedof_read_accel_mag(), "ninedof_read_accel_mag");
+  int x = 0;
+  int y = 0;
+  int z = 0;
+  check_err(ninedof_read_magnetometer_sync(&x, &y, &z), "ninedof_read_magnetometer_sync");
+  unsigned char sound_pres;
+  check_err(sound_pressure_read_sync(&sound_pres), "sound_pressure_read_sync");
+
+  // Analog inputs: P0-P2
+  uint16_t val = 0;
+  check_err(adc_sample_sync(0, &val), "adc_sample_sync");
+  int p0 = (val * 3300) / (4095 << 4);
+  check_err(adc_sample_sync(1, &val), "adc_sample_sync");
+  int p1 = (val * 3300) / (4095 << 4);
+  check_err(adc_sample_sync(2, &val), "adc_sample_sync");
+  int p2 = (val * 3300) / (4095 << 4);
+  if (p0 < 10 || p1 < 10 || p2 < 10) { // pin is connected to gnd
+    check_err(led_on(20), "led_on");
+  }
+
+  // Digital inputs: P8, P9, P16
+  int p8 = 0;
+  check_err(gpio_read(8, &p8), "gpio_read");
+  int p9 = 0;
+  check_err(gpio_read(9, &p9), "gpio_read");
+  int p16 = 0;
+  check_err(gpio_read(16, &p16), "gpio_read");
+
+  // print results
+  printf("[Micro:bit Sensor Reading]\n");
+  // Temp is given in hundredths of a degree C, in format XX00
+  printf("  Temperature: reading: %d.0 degrees C\n", temp / 100);

```

Comment:
When you run the app, do you get a temperature that varies or changes when you put a finger on the microcontroller?

Mine always seems to read 24.0 degrees, which is weird. I can't see how that would be the fault of this application.

Reply:
Looks like I just wasn't patient enough yesterday. Temperature works fine!